### PR TITLE
add feature: save poster locally

### PR DESCRIPTION
**Pull request by Marlon154**
This update introduces an automatic poster saving feature for the Moviegrabber plugin, allowing users to seamlessly download and save movie posters directly into their Obsidian vault. Alongside, we've refined the naming convention for both poster images and movie note files, utilizing the movie's title and year for a clear, consistent, and searchable format.

Automatic Poster Saving: Upon enabling this feature in the settings, the plugin will automatically download movie posters when creating new movie notes, saving them in a user-specified directory.

### CODE REVIEW

**Comment by Superschnizel** [comment]
This looks like a great addition. I see only one uncertainty/ issue.

> Alongside, we've refined the naming convention for both poster images and movie note files, utilizing the movie's title and year for a clear, consistent, and searchable format.

I dont see any edit in the way movie notes are named, am i correct that this is just an error in the description or did i miss that in the code?
Movie Note names can be created using a template, maybe it would be reasonable to apply that template to the image filename aswell?

**Review by Superschnizel — CHANGES_REQUESTED** [review]
I like this feature, i only think the images should use the filename specified by the template.

File: main.ts [hunk 1]
```diff
@@ -251,6 +251,12 @@ export default class Moviegrabber extends Plugin {
 	}
 
 	async createNote(item : MovieData, type : 'movie' | 'series', path : string, tFile : TFile | null=null) {
+		if (this.settings.enablePosterImageSave && item.Poster && item.Poster !== "N/A") {
+			// Construct the image name in a similar fashion to the provided example
+			const imageName = `${item.Title}_${item.Year}`.replace(/[^a-z0-9]+/gi, '_').toLowerCase() + '.jpg';
```
[the remark this answers is not in the document]
Comment (Marlon154) [reply main.ts]: Good point, I fixed it.

**Review by Superschnizel — COMMENTED** [review]
Ah, i see that we are doing the title templating twice now, it would probably be best to just pass the tamplated title along to the `createNote` function.

**Comment by Marlon154** [comment]
I moved it, I think this makes more sense. Also I added notifications for fail and success.

File: main.ts [hunk 2]
```diff
@@ -248,10 +249,27 @@ export default class Moviegrabber extends Plugin {
 		}
 
 		this.createNote(itemData, type, path);
```
Comment (Superschnizel) [inline main.ts:251]: i think this needs to be calles after the Posterimage save.

File: main.ts [hunk 3]
```diff
@@ -248,10 +249,27 @@ export default class Moviegrabber extends Plugin {
...
+
+		if (this.settings.enablePosterImageSave && itemData.Poster !== null && itemData.Poster !== "N/A") {
+			const imageName = `${cleanedTitle}.jpg`;
+			const posterDirectory = this.settings.posterImagePath;
+			this.downloadAndSavePoster(item.Poster, posterDirectory, imageName)
+			this.downloadAndSavePoster(item.Poster, posterDirectory, imageName)
```
Comment (Superschnizel) [inline main.ts:257]: does this need to be called twice?

**Review by Superschnizel — CHANGES_REQUESTED** [review]
Looks good, other than the two issues I commented.

File: main.ts (hunk 2 above)
Comment (Marlon154) [reply main.ts:251]: no doesn't matter

File: main.ts (hunk 3 above)
Comment (Marlon154) [reply main.ts]: no idea how this happend

File: main.ts (hunk 2 above)
Comment (Superschnizel) [reply main.ts:251]: but then the posterimage data will not be writen to the itemData and will not be available in the template or am i wrong?

Comment (Marlon154) [reply main.ts]: Misunderstanding

**Comment by Superschnizel** [comment]
I will test it and then create a new Version.

**Pull request merged**